### PR TITLE
[k8s-configuration] Release v1.2.0 with Flux v2 Support

### PR DESCRIPTION
**Pull request by jonathan-innis**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
k8s-configuration

File: src/dataprotection/azext_dataprotection/manual/version.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+
+VERSION = "0.2.0"
```
Comment (zhoxing-ms) [inline src/dataprotection/azext_dataprotection/manual/version.py:11]: Why do we need to modify this file?

File: src/k8s-configuration/azext_k8s_configuration/_params.py [hunk 2]
```diff
@@ -3,44 +3,134 @@
...
+                   help='Maximum time to reconcile the source before timing out')
+        c.argument('sync_interval',
+                   options_list=['--interval', '--sync-interval'],
+                   help='Time between reconciliations of the source on the cluster')
+        c.argument('branch',
+                   arg_group="Git Repo Ref",
+                   help='Branch within the git source to reconcile with the cluster')
+        c.argument('tag',
+                   arg_group="Git Repo Ref",
+                   help='Tag within the git source to reconcile with the cluster')
```
Comment (zhoxing-ms) [inline src/k8s-configuration/azext_k8s_configuration/_params.py:69]: May I ask is this the tag stored in ARM service? If so, do we need to consider using `tags_type`?

File: src/k8s-configuration/azext_k8s_configuration/_params.py [hunk 3]
```diff
@@ -3,44 +3,134 @@
...
+                   help='HTTPS username for private repository sync')
+        c.argument('https_key',
+                   arg_group="Auth",
+                   help='HTTPS token/password for private repository sync')
+        c.argument('https_ca_cert',
+                   arg_group="Auth",
+                   help='Base64-encoded HTTPS CA certificate for TLS communication with private repository sync')
+        c.argument('https_ca_cert_file',
+                   arg_group="Auth",
+                   help='Filepath to HTTPS CA certificate file for TLS communication with private repository sync')
```
Comment (zhoxing-ms) [inline src/k8s-configuration/azext_k8s_configuration/_params.py:93]:
```suggestion
                   help='File path to HTTPS CA certificate file for TLS communication with private repository sync')
```

File: src/k8s-configuration/azext_k8s_configuration/_params.py [hunk 4]
```diff
@@ -3,44 +3,134 @@
...
+        c.argument('known_hosts',
+                   arg_group="Auth",
+                   help='Base64-encoded known_hosts data containing public SSH keys required to access private Git instances')
+        c.argument('known_hosts_file',
+                   arg_group="Auth",
+                   help='Filepath to known_hosts contents containing public SSH keys required to access private Git instances')
```
Comment (zhoxing-ms) [inline src/k8s-configuration/azext_k8s_configuration/_params.py:99]:
```suggestion
                   help='File path to known_hosts contents containing public SSH keys required to access private Git instances')
```

File: src/k8s-configuration/azext_k8s_configuration/commands.py [hunk 5]
```diff
@@ -5,19 +5,47 @@
...
-        g.custom_command('update', 'update_k8s_configuration')
-        g.custom_command('delete', 'delete_k8s_configuration', confirmation=True)
-        g.custom_command('list', 'list_k8s_configuration', table_transformer=k8s_configuration_list_table_format)
-        g.custom_show_command('show', 'show_k8s_configuration', table_transformer=k8s_configuration_show_table_format)
+    with self.command_group('k8s-configuration flux', k8s_configuration_fluxconfig_sdk, client_factory=k8s_configuration_fluxconfig_client, is_preview=True) as g:
+        g.custom_command('create', 'flux_config_create', supports_no_wait=True)
+        g.custom_command('update', 'flux_config_update', supports_no_wait=True)
+        g.custom_command('list', "flux_config_list", table_transformer=fluxconfig_list_table_format)
+        g.custom_show_command('show', 'flux_config_show', table_transformer=fluxconfig_show_table_format)
+        g.custom_command('delete', 'flux_config_delete', supports_no_wait=True)
```
Comment (zhoxing-ms) [inline src/k8s-configuration/azext_k8s_configuration/commands.py:38]: Do we need to consider adding `confirmation=True` for the delete operation

File: src/k8s-configuration/azext_k8s_configuration/commands.py [hunk 6]
```diff
@@ -5,19 +5,47 @@
...
+
+    with self.command_group('k8s-configuration flux kustomization', k8s_configuration_fluxconfig_sdk, client_factory=k8s_configuration_fluxconfig_client, is_preview=True) as g:
+        g.custom_command('create', 'flux_config_create_kustomization', supports_no_wait=True)
+        g.custom_command('update', 'flux_config_update_kustomization', supports_no_wait=True)
+        g.custom_command('delete', 'flux_config_delete_kustomization', supports_no_wait=True)
```
Comment (zhoxing-ms) [inline src/k8s-configuration/azext_k8s_configuration/commands.py:43]: Same as above

File: src/k8s-configuration/azext_k8s_configuration/_params.py (hunk 2 above)
Comment (jonathan-innis) [reply src/k8s-configuration/azext_k8s_configuration/_params.py:69]: This is not an ARM tag, this is a different thing. This is referring to a tag in the Git repository

File: src/k8s-configuration/azext_k8s_configuration/commands.py (hunk 5 above)
Comment (jonathan-innis) [reply src/k8s-configuration/azext_k8s_configuration/commands.py:38]: We are using custom delete confirmation, so we are not using the default confirmation as provided by the CLI

https://github.com/AzureArcForKubernetes/azure-cli-extensions/blob/k8s-configuration-release-1.2.0/src/k8s-configuration/azext_k8s_configuration/providers/FluxConfigurationProvider.py#L341

File: src/k8s-configuration/azext_k8s_configuration/commands.py (hunk 6 above)
Comment (jonathan-innis) [reply src/k8s-configuration/azext_k8s_configuration/commands.py:43]: Same comment as above, we are using custom confirmation

File: src/k8s-configuration/azext_k8s_configuration/commands.py (hunk 5 above)
Comment (zhoxing-ms) [reply src/k8s-configuration/azext_k8s_configuration/commands.py:38]: Got it, thanks~

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**